messaging/src/main/java/org/axonframework/common/infra/FilesystemStyleComponentDescriptor.java

```diff
@@ -0,0 +1,377 @@
+        properties.put(name, value);
+    }
+
+    @Override
+    public void describeProperty(@Nonnull String name, Boolean value) {
+        properties.put(name, value);
+    }
+
+    @Override
+    public String describe() {
+        try {
+            return new TreeRenderer().render(properties);
+        } catch (Exception e) {
+            throw new ComponentDescriptorException(
+                    "Error generating Filesystem style description",
+                    e
+            );
+        }
+    }
+
+    private record SymbolicLink(String targetPath) {
+
+        private static final String SYMLINK_INDICATOR = " -> ";
+
+        @Override
+        public String toString() {
+            return SYMLINK_INDICATOR + targetPath;
+        }
+    }
+
+    private static class TreeRenderer {
+
+        private static final String CORNER = "└── ";
+        private static final String TEE = "├── ";
+        private static final String VERTICAL = "│   ";
+        private static final String SPACE = "    ";
+
+        private final StringBuilder result = new StringBuilder();
+
+        String render(Map<String, Object> properties) {
+            result.append(ROOT_PATH).append("\n");
+            var context = new RenderContext("", "");
+            render(properties, context);
+            return result.toString();
+        }
+
+        private void render(Map<String, Object> properties, RenderContext context) {
+            var entries = new ArrayList<>(properties.entrySet());
+
+            for (int i = 0; i < entries.size(); i++) {
+                var entry = entries.get(i);
+                var name = entry.getKey();
+                var value = entry.getValue();
+
+                var isLast = (i == entries.size() - 1);
+                var connector = isLast ? CORNER : TEE;
+
+                renderProperty(name, value, connector, context, isLast);
+            }
+        }
+
+        private void renderProperty(
+                String name,
+                Object value,
+                String connector,
+                RenderContext context,
+                boolean isLastInCollection
+        ) {
+            switch (value) {
+                case FilesystemStyleComponentDescriptor descriptor -> renderComponentDirectory(name,
+                                                                                               descriptor,
+                                                                                               context,
+                                                                                               isLastInCollection);
+                case List<?> list -> renderList(name, list, context, isLastInCollection);
+                case Map<?, ?> map -> renderMap(name, map, context, isLastInCollection);
+                case SymbolicLink link -> renderSymlink(name, link, connector, context);
+                case null, default -> renderSimpleValue(name, value, connector, context);
```

Comment:
Nit: I think I would personally pass along the `isLastInCollection` to all four methods. Now, there's a mix between the `isLast` boolean and the `connector`, even though the `connector` would (at least partially) be usable for the `renderComponentDirectory`/`renderList`/`renderMap` method as well. 

By enforcing the `isLast` on the `renderSymlink` and `renderSimpleValue` too, all solutions would align.

Reply:
Now should be ok! :)